### PR TITLE
Bug 1865839: daemon: cleanup pending deployment on an error

### PR DESCRIPTION
**Pull request by sinnykumari**
We handle rpm-ostree operations such as OS update,
karg update, kernel switch and additional package
overlay in applyOSChanges(). When an error occurs
during any one of these operations, we would want
to rollback any changes we have performed on booted
OS so that we can cleanly retry again to apply
the changes. rpm-ostree provides us `cleanup -p`
options which removes pending deployment.

### CODE REVIEW

File: pkg/daemon/update.go [hunk 1]
```diff
@@ -306,7 +306,17 @@ func ExtractOSImage(imgURL string) (osImageContentDir string, err error) {
...
 
-func (dn *Daemon) applyOSChanges(oldConfig, newConfig *mcfgv1.MachineConfig) error {
+// Remove pending deployment on OSTree based system
+func removePendingDeployment() error {
+	args := []string{"cleanup", "-p"}
+	_, err := runGetOut("rpm-ostree", args...)
+	if err != nil {
+		return err
+	}
+	return nil
```
Comment (runcom) [inline pkg/daemon/update.go:316]: nit you can `return err` directly if you don't plan to wrap it

**Comment by runcom** [comment]
💯 thanks Sinny for the quick fix!

File: pkg/daemon/update.go (hunk 1 above)
Comment (sinnykumari) [reply pkg/daemon/update.go]: right, fixed.

**Pull request merged**